src/presign/participant.rs

```diff
@@ -329,69 +331,62 @@ impl PresignParticipant {
     fn gen_round_one_msgs<R: RngCore + CryptoRng>(
         &mut self,
         rng: &mut R,
-        message: &Message,
+        sid: Identifier,
         input: &Input,
-    ) -> Result<ProcessOutcome<<Self as ProtocolParticipant>::Output>> {
+    ) -> Result<Vec<Message>> {
         info!("Generating round one presign messages.");
 
-        // Reconstruct keyshare and other participants' public keyshares from local
-        // storage
+        // Reconstruct keyshare info from external input.
         let keyshare = get_keyshare(self.id, input)?;
         let other_public_auxinfo = input.all_but_one_auxinfo_public(self.id);
 
-        // Run Round One
+        // Run round one.
         let (private, r1_publics, r1_public_broadcast) =
             keyshare.round_one(rng, &self.retrieve_context(), &other_public_auxinfo)?;
 
-        // Store private round one value locally
+        // Store private round one value locally.
         self.local_storage
             .store::<storage::RoundOnePrivate>(self.id, private);
 
-        // Publish public round one value to all other participants on the channel
-        let non_broadcast_messages = r1_publics
+        // Publish public round one values to all other participants on the channel.
+        let mut messages = r1_publics
             .into_iter()
             .map(|(other_id, r1_public)| {
                 Ok(Message::new(
                     MessageType::Presign(PresignMessageType::RoundOne),
-                    message.id(),
+                    sid,
                     self.id,
                     other_id,
                     &serialize!(&r1_public)?,
                 ))
             })
             .collect::<Result<Vec<_>>>()?;
-
         let broadcast_messages = self.broadcast(
             rng,
             MessageType::Presign(PresignMessageType::RoundOneBroadcast),
             serialize!(&r1_public_broadcast)?,
-            message.id(),
+            sid,
             BroadcastTag::PresignR1Ciphertexts,
         )?;
-
-        // Additionally, handle any round 1 messages which may have been received too
-        // early
```

Comment:
I think we still need this check.

I'm concerned about the case where we have not completed `ready`, we get a r1 broadcast and regular message from another party. In that case we'd do the whole r1 for those messages -- there's no check in the `handle_round_one` message functions to see if we finished the ready round.

Ready stuff is slightly weird because it's not in the protocol description, but I think what happens in the code is that every party self-initializes, then sends a "ready" message to every other party, and we only start once we've gotten all of them. We might want to stop doing that, but I'd rather remove it in a separate PR.

Reply:
I think that's okay: we can handle round one messages before we ourselves have generated our round one message (which is really what `ready` seems to be all about: making sure all parties are ready before we generate messages to send).

This is also how the other protocols (`auxinfo` and `keygen`) work (not that that means it's correct, just that if this really is an issue here we'll need to address it in those protocols as well)

Reply:
Oh, I see, thanks for clarifying.